tests/GithubApp.py

```diff
@@ -99,8 +102,18 @@ def testGetPublicApp(self):
         self.assertEqual(app.url, "/apps/github-actions")
 
     def testGetAuthenticatedApp(self):
-        # For this to work correctly in record mode, this test must be run with --auth_with_jwt
-        app = self.g.get_app()
+        auth = github.Auth.AppAuth(APP_ID, PRIVATE_KEY)
+        g = github.Github(auth=auth)
+
+        with self.assertWarns(DeprecationWarning) as warning:
+            app = g.get_app()
+
+            self.assertWarningIn(
```

Comment:
To be honest, I don't like `assertWarningIn`, because it may hide unexpected warnings that we want to fail on.

Having `httpretty` warn here is annoying, but we can disable that and everything works as expected:

```suggestion
        with self.assertWarns(DeprecationWarning) as warning:
            # we ignore warnings from httpretty dependency
            import warnings
            warnings.filterwarnings("ignore", module="httpretty")

            app = g.get_app()

            self.assertWarning(
```

Reply:
oh great, I did not know we could filter the warnings. Will change. Thanks